content/docs/product/managed-apps/dev-experience/git-subtree.md

```diff
@@ -409,4 +409,34 @@ In general, we have two options here:
    git commit -am "applied custom changes"
    # you're ready continue with updating to the current state
    ```
+## Fixing when original subtree-split commit is lost
 
+For reasons we're not completely sure of (but that for sure include history rewrite with `git push --force`), it might happen that the commit ID that is merged as a subtree from the artificial sub-directory tree created with `git subtree split` might change its commit ID, even though the set of changes is still the same. One way you can get out of this situation is:
+
+1) Figure out (from normal git log) in the `main` branch (or the working branch) when we did the last actual merge from upstream. This merge commit has the following message in it:
+
+```
+Squashed 'your/subdir' content from commit [short ID]
+    
+git-subtree-dir: your/subdir
+git-subtree-split: [long ID]
+```
+
+Let's assume in our case the commit was done on 03.01.2023.
+
+2) Do normal git subtree split, then checkout this tree so you can inspect its log. Start with checkout `git checkout temp-split-branch` then use `git log`. Now find the last artificial commit ID that we merged - its hash is different now, but we assume the set of patches is the same. In our case we look for the newest commit older than 03.01.2023 in the artificial history created by `subtree split`. Let's assume this commit has ID `5a9c69f11f1466569e04c8a60cbb132617d2185f`.
```

Comment:
Sorry, I didn't really use the `git subtree` feature much.

Reading this makes me feel `git submodule` (possibly pointing to an automatically synchronized fork) could be a more sane approach.